crates/client/rpc-core/src/lib.rs

```diff
@@ -135,4 +135,8 @@ pub trait StarknetRpcApi {
     /// Returns the receipt of a transaction by transaction hash.
     #[method(name = "getTransactionReceipt")]
     fn get_transaction_receipt(&self, transaction_hash: FieldElement) -> RpcResult<MaybePendingTransactionReceipt>;
+
+    /// Checks if L1 Message Nonce has not been used
+    #[method(name = "l1NonceUnused")]
+    fn l1_nonce_unused(&self, nonce: u64) -> RpcResult<bool>;
```

Comment:
Why do we need to expose this to external users as an RPC route?

Reply:
Not needed in this implementation, will remove it.